baremetal/metal3machine_manager.go

```diff
@@ -782,7 +782,13 @@ func (m *MachineManager) chooseHost(ctx context.Context) (*bmh.BareMetalHost, *p
 			helper, err := patch.NewHelper(&hosts.Items[i], m.client)
 			return &hosts.Items[i], helper, err
 		}
-		if !host.Available() {
+		if host.Spec.ConsumerRef != nil {
+			continue
+		}
+		if host.GetDeletionTimestamp() != nil {
+			continue
+		}
+		if host.HasError() {
```

Comment:
Maybe we shouldn't use this convenience method either.

Reply:
Done